base/ltfinal.dtx

```diff
@@ -1077,84 +1075,61 @@
 %    And whilst we're doing things with uc/lc tables, here are two
 %    commands to upper- and lower-case a string.
 %
-%    \emph{Note} that this implementation is subject to change!  At
-%    the moment we're not providing any way to extend the list of
-%    uc/lc commands, since finding a good interface is difficult.
-%    These commands have some nasty features, such as uppercasing
-%    mathematics, environment names, labels, etc.  A much better
-%    long-term solution is to use all-caps fonts, but these aren't
-%    generally available.
 %
+% \changes{v2.1h}{2019/09/14}{Expand UTF8 chars when case changing (github/177)}
+% \changes{v2.2r}{2022/02/20}
+%                {Use \cs{@expl@text@uppercase@@n}, removing local redefinition of \cs{UTF@two@octets@noexpand}}%
+% \changes{v2.2t}{2022/04/21}{Support \cs{noexpand} in argument of \cs{@expl@text@uppercase@@n}}
+% \changes{v2.2v}{2022/026/30}{Just use \cs{text_lowercase:n} without \cs{protectd@edf} gh/881x}%
 % Two wrappers around the L3 case changing functions.
 % |\protected| to make them mostly safe as replacements for |uppercase|
 % and |\lowercase|.  
+%
+% In
+%\begin{verbatim}
+%    \markboth{\MakeUppercase\contentsname}
+%             {\MakeUppercase\contentsname}
+%\end{verbatim}
+%    then the uppercasing is only done to the first letter of the
+%    contents name, since the mark expands out to:
+%\begin{verbatim}
+%    \mark{\protect\MakeUppercase Table of Contents}
+%         {\protect\MakeUppercase Table of Contents}
+%\end{verbatim}
+%    In order to get round this, we redefine |\MakeUppercase| and
+% |\MakeLowercase| to grab their argument and brace it. 
+%
+% Earlier versions needed to procsss |\@uclclist| in an |\edef|
+% to handle legacy input encodings, but recent (2022) expl3 versions
+% handle non-UTF8 text natively so we simply call the |\text_...case:n|
+% functions.
 %    \begin{macrocode}
 \ExplSyntaxOn
-\protected\def\@expl@text@lowercase@@n{\text_lowercase:n}
-\protected\def\@expl@text@uppercase@@n{\text_uppercase:n}
+\protected\def\MakeUppercase#1{\text_uppercase:n{#1}}
+\protected\def\MakeLowercase#1{\text_lowercase:n{#1}}
 %    \end{macrocode}
 %
 % \changes{v2.2u}{2022/06/02}{Add \cs{NoCaseChange}}
+% \changes{v2.2uv}{2022/06/02}{Add \cs{AddToNoCaseChangeList}}
 % |\NoCaseChange| protects its argument from the case change functions.
+%
+% |\AddToNoCaseChangeList| Allows new commands to protect their arguments, eg
+% |AddToNoCaseChangeList{\eqref}| would protect the argument of |\eqref|
+% in the same way as the argument of |\ref|.
 %    \begin{macrocode}
+\def\AddToNoCaseChangeList{\tl_put_right:Nn \l_text_case_exclude_arg_tl}
```

Comment:
Should likely be `\protected`

Reply:
I wondered about that but thought it would never be used in an expansion context, but I guess you are right in which case I may as well use an l3 defininition form since it's in an l3 syntax area anyway

Reply:
done c31117a3